### ISSUE TITLE
Running a task slow via agent in build definition vs running directly on machine.

### ISSUE THREAD

**Issue by sunilsurana** [microsoft/azure-pipelines-agent#2629]
**Agent Version and Platform**
Agent 2.160.1 running on Windows Server Server 2016

**What's the issue?**
I have registered my dev box as an agent in one of our self hosted pool
We have a task which complies typescript files. This is a task invoked in build pipeline.
When i run build definition this task takes 7 minutes to run
If i run this task directly on my local dev box in same repo it takes 4 minutes (There is no caching involved here)

Other info

This is a process intensive task and multiple typescript files are compiled in parallel.

**Comment by jtpetty** [comment]
@sunilsurana - I am not sure what action you would like taken on this issue. It seems that your dev box is more optimized for compiling typescript files than the hosted agents we provide. We are consistently working to improve our hosted agent offerings, but you are always welcome to self host agents for improved performance for your work load.

**Comment by jtpetty** [comment]
Closing as there is not an actionable issue. If there is something else we should be looking, please open a new issue.

**Comment by sunilsurana** [comment]
Hi @jtpetty 

I am fine with closing of this issue till I am not able to provide reproducible steps.

But again the issue is not running on dev box vs agent. The issue is when i register my dev box as agent in private pool and run a job via agent on my dev box it is very slow compared to running it directly on dev box
